src/ncp/ncp_base.cpp

```diff
@@ -4716,6 +4729,142 @@ ThreadError NcpBase::SetPropertyHandler_MAC_WHITELIST_ENABLED(uint8_t header, sp
+                   );
+
+    if (parsedLength > 0)
+    {
+        errorCode = otLinkRawSrcMatchEnable(mInstance, isEnabled);
+        if (errorCode == kThreadError_None)
+        {
+            errorCode = HandleCommandPropertyGet(header, key);
+        }
+        else
+        {
+            errorCode = SendLastStatus(header, ThreadErrorToSpinelStatus(errorCode));
+        }
+    }
+    else
+    {
+        errorCode = SendLastStatus(header, SPINEL_STATUS_PARSE_ERROR);
+    }
+
+    return errorCode;
+}
+
+ThreadError NcpBase::SetPropertyHandler_MAC_SRC_MATCH_SHORT_ADDRESSES(uint8_t header, spinel_prop_key_t key,
+                                                                      const uint8_t *value_ptr, uint16_t value_len)
+{
+    ThreadError errorCode = kThreadError_None;
+    const uint8_t *data = value_ptr;
+    uint16_t data_len = value_len;
+
+    // Clear the list first
+    SuccessOrExit(errorCode = otLinkRawSrcMatchClearShortEntries(mInstance));
+
+    // Loop through the addresses and add them
+    while (data_len >= sizeof(uint16_t))
+    {
+        spinel_ssize_t parsedLength;
+        uint16_t short_address;
+
+        parsedLength = spinel_datatype_unpack(
+            data,
+            data_len,
+            "S",
+            &short_address
+        );
+
+        VerifyOrExit(parsedLength > 0, errorCode = kThreadError_Parse);
+
+        data += parsedLength;
+        data_len -= (uint16_t)parsedLength;
+
+        SuccessOrExit(errorCode = otLinkRawSrcMatchAddShortEntry(mInstance, short_address));
+    }
+
+    (void)SendPropertyUpdate(
+            header,
+            SPINEL_CMD_PROP_VALUE_IS,
+            key,
+            value_ptr,
+            value_len
+        );
+
+exit:
+
+    if (errorCode != kThreadError_None)
+    {
+        errorCode = SendLastStatus(header, ThreadErrorToSpinelStatus(errorCode));
+    }
+
+    return errorCode;
+}
+
+ThreadError NcpBase::SetPropertyHandler_MAC_SRC_MATCH_EXTENDED_ADDRESSES(uint8_t header, spinel_prop_key_t key,
+                                                                         const uint8_t *value_ptr, uint16_t value_len)
+{
+    ThreadError errorCode = kThreadError_None;
+    const uint8_t *data = value_ptr;
+    uint16_t data_len = value_len;
+
+    // Clear the list first
+    SuccessOrExit(errorCode = otLinkRawSrcMatchClearExtEntries(mInstance));
+
+    // Loop through the addresses and add them
+    while (data_len >= sizeof(otExtAddress))
+    {
+        spinel_ssize_t parsedLength;
+        uint8_t *ext_address;
+
+        parsedLength = spinel_datatype_unpack(
+            data,
+            data_len,
+            "E",
+            &ext_address
+        );
+
+        VerifyOrExit(parsedLength > 0, errorCode = kThreadError_Parse);
+
+        data += parsedLength;
+        data_len -= (uint16_t)parsedLength;
+
+        SuccessOrExit(errorCode = otLinkRawSrcMatchAddExtEntry(mInstance, ext_address));
+    }
+
+    (void)SendPropertyUpdate(
```

Comment:
Let's handle and return error code from `SendPropertyUpdate` (particularly for running out of buffer) like other hanlders. The reason behind this is that `HandleReceive()` will be able to notify the host later that NCP ran out of buffer during an operation. 

`InsertPropertyHandler_IPV6_ADDRESS_TABLE` is a good example of doing this where we use `VerifyOrExit` and handle some of the erros at `exit` label. 

Other way of doing this is like `SetPropertyHandler_NET_MASTER_KEY` which uses a sequence of `if`s...